### PR TITLE
Managed dns

### PR DESCRIPTION
**Pull request by grollum**
Opening this already to receive feedback on the backend part.
When running wirtbot with env `MANAGED_DNS_ENABLED=1|true` a new endpoint `/update-devices` accepts a message 
```
{
  "signature": "....",
  "message": "1.2.3.4 some-device.my-network.home\n5.6.7.8 some-other-device.my-network.home"
}
```
and updates a file on the wirtbot host with the contents of message.message. The file is `/etc/wirt_devices` by default and can be controlled with the `MANAGED_DNS_DEVICE_FILE` env var.

### CODE REVIEW

File: WirtBot/src/main.rs [hunk 1]
```diff
@@ -164,6 +170,43 @@ fn update(
         .map(|_| format!("Config updated"))
 }
 
+fn update_devices(
```
Comment (b-m-f) [inline WirtBot/src/main.rs:173]: Can you rename this to `update_device_dns_entries`?

File: WirtBot/src/main.rs [hunk 2]
```diff
@@ -164,6 +170,43 @@ fn update(
...
+    public_key: PublicKey,
+) -> impl Filter<Extract = (String,), Error = warp::Rejection> + Copy {
+    warp::post()
+        .and(warp::path("update-devices"))
```
Comment (b-m-f) [inline WirtBot/src/main.rs:177]: Same for the endpoint name -> `update_device_dns_entries`

File: WirtBot/src/managed_dns.rs [hunk 3]
```diff
@@ -0,0 +1,56 @@
+use std::env;
+
+const MANAGED_DNS_FEATURE_FLAG :&str = "MANAGED_DNS_ENABLED";
```
Comment (b-m-f) [inline WirtBot/src/managed_dns.rs:3]: Nice one. Ill try to adapt the other flags to use consts like this after your PR as well!

File: WirtBot/src/managed_dns.rs [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    #[test]
+    fn test_enabled() {
+        env::set_var(MANAGED_DNS_FEATURE_FLAG, "true");
+        assert!(enabled());
+        env::set_var(MANAGED_DNS_FEATURE_FLAG, "1");
+        assert!(enabled())
+    }
+
+    #[test]
+    #[should_panic]
```
Comment (b-m-f) [inline WirtBot/src/managed_dns.rs:40]: Nice, I didnt know about this!

**Comment by grollum** [comment]
I would like to provide a dnsmasq service together with wirtbot via docker-compose for dev. May I add this to the `dev.yml` compose file or would you prefer a dedicated one?
Also: How about a docker-compose file in project root running all things together?

**Comment by b-m-f** [comment]
Id prefer a copy of `dev.yml`. Maybe `wirtbot_with_dns.yml`?

**Comment by b-m-f** [comment]
Hey @grollum, hows it going?

I will be able to work on Wirt this weekend. Anything I can help you with?

**Comment by b-m-f** [comment]
To get this into master I will start work on your branch. Hope u dont mind

**Comment by b-m-f** [comment]
New PR now over here https://github.com/b-m-f/Wirt/pull/34

**Pull request closed without merging**